--- configure.in.orig	2009-06-24 01:37:32.000000000 -0400
+++ configure.in	2009-06-24 01:37:51.000000000 -0400
@@ -1036,8 +1036,6 @@
                 [xbmc/visualizations/OpenGLSpectrum/Makefile]
                 [xbmc/visualizations/WaveForm/Makefile]
                 [xbmc/visualizations/iTunes/Makefile]
-                [tools/Linux/xbmc.sh]
-                [tools/XBMCTex/Makefile]
                 [tools/EventClients/Clients/OSXRemote/Makefile]
 )
 
