--- xbmc/utils/ArabicShaping.h.orig	2009-05-03 10:58:25.000000000 -0400
+++ xbmc/utils/ArabicShaping.h	2009-06-24 00:20:17.000000000 -0400
@@ -25,11 +25,7 @@
  *
  */
 
-#ifndef _LINUX
 #include "lib/libfribidi/fribidi.h"
-#else
-#include <fribidi/fribidi.h>
-#endif
 
 /**
  * Shapes an Arabic text chunk

Index: tools/XBMCTex/XBMCTex.cpp
===================================================================
--- tools/XBMCTex/XBMCTex.cpp	(revision 20692)
+++ tools/XBMCTex/XBMCTex.cpp	(working copy)
@@ -648,7 +648,7 @@
 			OutputFilename = args[++i];
 #ifdef _LINUX
       char *c = NULL;
-      while ((c = strchr(OutputFilename, '\\')) != NULL) *c = '/';
+      while ((c = (char *)strchr(OutputFilename, '\\')) != NULL) *c = '/';
 #endif
 		}
     else if (!stricmp(args[i], "-noprotect") || !stricmp(args[i], "-p"))


